cmd/syncthing/connections.go

```diff
@@ -353,3 +353,24 @@ func (s *connectionSvc) shouldLimit(addr net.Addr) bool {
 	}
 	return !tcpaddr.IP.IsLoopback()
 }
+
+func (s *connectionSvc) VerifyConfiguration(from, to config.Configuration) error {
+	return nil
+}
+
+func (s *connectionSvc) CommitConfiguration(from, to config.Configuration) bool {
```

Comment:
I just meant comparing against existing devices was easier to follow, not some sort of optimizatio that I had in mind.

Probably doesn't compile:
```
existingDevices := make(map[protocol.DeviceID]struct{}, len(from.Devices))
for _, dev := range from.Devices {
    existingDevices[dev.DeviceID] = struct{}{}
}

for _, dev := range to.Devices {
    if _, ok := !existingDevices[dev.DeviceID]; !ok {
        return false
    }
}
return true
```